transformer/releasetransformer.go

```diff
@@ -0,0 +1,260 @@
+}
+
+// Structs representing the raw elastic search response
+
+type ESReleaseResponse struct {
+	Took     int                      `json:"took"`
+	TimedOut bool                     `json:"timed_out"`
+	Hits     ESReleaseResponseSummary `json:"hits"`
+}
+
+type ESReleaseResponseSummary struct {
+	Total int                    `json:"total"`
+	Hits  []ESReleaseResponseHit `json:"hits"`
+}
+
+type ESReleaseResponseHit struct {
+	Source    ESReleaseSourceDocument `json:"_source"`
+	Highlight ESReleaseHighlight      `json:"highlight"`
+}
+
+type ESReleaseSourceDocument struct {
+	URI         string       `json:"uri"`
+	DateChanges []dateChange `json:"dateChanges,omitempty"`
+
+	Description struct {
+		Title              string          `json:"title"`
+		Summary            string          `json:"summary"`
+		ReleaseDate        string          `json:"releaseDate,omitempty"`
+		Published          bool            `json:"published"`
+		Cancelled          bool            `json:"cancelled"`
+		Finalised          bool            `json:"finalised"`
+		Topics             []string        `json:"topics"`
+		NationalStatistic  bool            `json:"nationalStatistic,omitempty"`
+		Keywords           []string        `json:"keywords,omitempty"`
+		NextRelease        string          `json:"nextRelease,omitempty"`
+		CancellationNotice []string        `json:"cancellationNotice"`
+		ProvisionalDate    string          `json:"provisionalDate"`
+		Edition            string          `json:"edition,omitempty"`
+		DatasetID          string          `json:"datasetId,omitempty"`
+		LatestRelease      bool            `json:"latestRelease"`
+		MetaDescription    string          `json:"metaDescription,omitempty"`
+		Language           string          `json:"language,omitempty"`
+		Source             string          `json:"source,omitempty"`
+		Contact            *releaseContact `json:"contact,omitempty"`
+	} `json:"description"`
+}
+
+type dateChange struct {
+	PreviousDate string `json:"previousDate,omitempty"`
+	ChangeNotice string `json:"changeNotice,omitempty"`
+}
+
+type ESReleaseHighlight struct {
+	DescriptionTitle     []string `json:"description.title"`
+	DescriptionEdition   []string `json:"description.edition"`
+	DescriptionSummary   []string `json:"description.summary"`
+	DescriptionMeta      []string `json:"description.metaDescription"`
+	DescriptionKeywords  []string `json:"description.keywords"`
+	DescriptionDatasetID []string `json:"description.datasetId"`
+}
+
+func NewReleaseTransformer() *ReleaseTransformer {
+	highlightReplacer := strings.NewReplacer("<em class=\"highlight\">", "", "</em>", "")
+	return &ReleaseTransformer{
+		higlightReplacer: highlightReplacer,
+	}
+}
+
+// TransformSearchResponse transforms an elastic search response into a structure that matches the v1 api specification
+func (t *ReleaseTransformer) TransformSearchResponse(_ context.Context, responseData []byte, _ string, highlight bool) ([]byte, error) {
```

Comment:
(question): why is these parameters ignored ? I assume this is done to match a specific api's function signature, so can this be  ```TransformSearchResponse(responseData []byte string, highlight bool) ([]byte, error)``` instead and refactoring the api as well if needed ?

Reply:
Yes the signature matches the ResponseTransformer interface and ensures the ReleaseTransformer implements the interface - this interface is used in the current search and I didn't want to touch or change that work, so I implement the interface although in thais case some of the parameters are not used